qa/sc_ft_limit_fork.py

```diff
@@ -0,0 +1,148 @@
+        assert_equal(we0_cert["cert"]["ftScFee"], 0)
+
+        # switch to the next consensus epoch
+        generate_next_block(sc_node, "first node", force_switch_to_next_epoch=True)
+        forging_info = http_block_forging_info(sc_node)
+        assert_equal(forging_info["bestEpochNumber"], 3)
+        check_wallet_coins_balance(sc_node, 100.00000050)
+
+        # before new cert is created, it should be possible to do Ft with 50 satoshi
+        self.do_ft(_50cent)
+
+        # generate new cert, it should contain new value for ftScFee = 54 satoshi
+        self.generate_certificate()
+        we0_cert_hash = mc_node.getrawmempool()[0]
+        print("Withdrawal epoch 0 certificate hash = " + we0_cert_hash)
+        we0_cert = mc_node.getrawtransaction(we0_cert_hash, 1)
+        assert_equal(we0_cert["cert"]["ftScFee"], Decimal(54) / Decimal(100000000))
+
+        # get to the next withdrawal epoch to enable new certificate
+        self.sync_all()
+        mc_node.generate(9)
+        generate_next_block(sc_node, "first node", force_switch_to_next_epoch=True)
+        mc_node.generate(1)
+        generate_next_blocks(sc_node, "first node", 1)
+        # try to do another FT with 50 satoshi, assert it fails
+        try:
+            self.do_ft(_50cent)
+        except Exception as e:
+            assert_equal('16: bad-sc-tx-not-applicable', e.error['message'])
+
+        # 55 satoshi and above should be successful
+        self.do_ft(0.00000055)
+        mc_node.generate(1)
+        generate_next_blocks(sc_node, "first node", 1)
+        check_wallet_coins_balance(sc_node, 100.00000155)
+
+    def generate_certificate(self):
+        mc_node = self.nodes[0]
+        sc_node = self.sc_nodes[0]
+        mc_node.generate(9)
+        generate_next_blocks(sc_node, "first node", 1)
+        mc_node.generate(1)
+        generate_next_blocks(sc_node, "first node", 1)
+
+        time.sleep(10)
+        while mc_node.getmempoolinfo()["size"] == 0 and sc_node.submitter_isCertGenerationActive()["result"]["state"]:
+            print("Wait for certificate in mc mempool...")
+            time.sleep(2)
+            sc_node.block_best()  # just a ping to SC node.
+            # For some reason, STF can't request SC node API after a while idle.
+        assert_equal(1, mc_node.getmempoolinfo()["size"], "Certificate was not added to Mc node mempool.")
+
+    def do_ft(self, amount):
+        sc_node = self.sc_nodes[0]
+        mc_node = self.nodes[0]
+
+        mempool_size_init = mc_node.getmempoolinfo()["size"]
+        sc_address = sc_node.wallet_createPrivateKey25519()["result"]["proposition"]["publicKey"]
+        ft_amount = amount
+        mc_return_address = mc_node.getnewaddress()
+
+        ft_args = [{
+            "toaddress": sc_address,
+            "amount": ft_amount,
+            "scid": self.sc_nodes_bootstrap_info.sidechain_id,
+            "mcReturnAddress": mc_return_address
+        }]
+        mc_node.sc_send(ft_args)
```

Comment:
Let's genrate mc and sc blocks and check that funds arrived to sidechain wallet.

Reply:
Discussed offline. Checks performed in run_test.